pulsar-client/src/main/java/org/apache/pulsar/client/impl/ZeroQueueConsumerImpl.java

```diff
@@ -81,6 +126,17 @@ public ZeroQueueConsumerImpl(PulsarClientImpl client, String topic, ConsumerConf
         return future;
     }
 
+    private Message<T> doZeroQueue() {
+        Message<T> result = null;
+        try {
+            result = internalReceive();
```

Comment:
internalReceive will hold a lock, we cannot call it directly when receiving message with timeout.

Reply:
I think a semaphore is needed here to complete the blocking